### ISSUE TITLE
Phoenix delivering incomplete assets

### ISSUE THREAD

**Issue by mmcc** [phoenixframework/phoenix#1500]
I know this sounds insane and I have this terrible fear that no one will be able to replicate the problem, but here we go.

I've got an entirely clean `web/static/js` directory at this point, with app.js only importing the `phoenix_html` helpers. If I save the file so it builds, then check the contents of `priv/static/js/app.js`, it looks perfectly normal. I can add a `console.log` or something along those lines, save, check that file and again, everything is _totally normal_ and as expected.

However, if I load that file in the browser via Phoenix, it just abruptly stops. To illustrate, here's the last ~10 lines of each file.

### $ cat `priv/static/js/app.js`

```
;require.register("web/static/js/app", function(exports, require, module) {
"use strict";

require("deps/phoenix_html/web/static/js/phoenix_html");

console.log('I think I must be going insane');
});

require('web/static/js/app');
//# sourceMappingURL=app.js.map
```

### GET localhost:4000/js/app.js

```
                                    }, {
                                        key: "setTimeout",
                                        value: function(_setTimeout) {
                                            function setTimeout() {
                                                return _setTimeout.apply(this, arguments);
                                            }

                                            setTimeout.toString = function() {
                                                return _set
```

I left the indentation in there to illustrate that it _literally_ just stops out of nowhere. If I change the source file and compile, it'll abruptly stop in different places, for example if I remove the `console.log` in the above example the cliff just moves to `setTime` a line earlier.

Again, I know this sounds crazy, but I've tried everything I can think to debug this and nothing has functionally changed the outcome. I saw a few issues around Brunch versions, but everything I see points to the correct file being written, it's just not being served. Also, just to make sure this wasn't something weird with the file headers I set up a static server in the directory and that also delivered the complete file as I would expect. Any help debugging this would be greatly appreciated. Thanks!

**Comment by josevalim** [comment]
Does it happen in multiple browsers? What is your brunch and phoenix versions?

**Comment by mmcc** [comment]
It's happened in every browser I've tested it on (Chrome, Safari, and Firefox) and just for the sake of my own sanity, cURL as well.

Brunch: 2.2.1
Phoenix: 1.1.4

**Comment by josevalim** [comment]
Given your versions, I don't expect anything weird to be happening. :S @mmcc do you have a sample app that reproduces the error? I would gladly take a look at it.

**Comment by mmcc** [comment]
So I'm using Docker / Docker Compose for my development environment. After sleeping on it, I took a step back and started inspecting the generated files outside of my editor and noticed some odd characters getting appended to the end of the file. That got me thinking, so I took a step back, cloned the repo into a new folder, built everything and ran directly from the host and...no problems. Somehow Docker is managing to cause this issue, and I have no clue how, but it's clearly not a Phoenix problem.

Really sorry for the false alarm, I should have removed Docker from the equation before opening this. I'm at a loss for what's actually going on, but that's clearly a question for a different tracker. Thanks for your time anyway, really appreciate you taking a look.
